jaxrs-api/src/main/java/javax/ws/rs/sse/SseEventSource.java

```diff
@@ -53,14 +53,20 @@
  * received SSE event {@code retry} field values set by the endpoint and adjusts the reconnect delay accordingly,
  * using the last received {@code retry} field value as the reconnect delay.
  * <p>
- * In addition to handling the standard connection loss failures, JAX-RS {@code SseEventSource} automatically deals with any
- * {@code HTTP 503 Service Unavailable} responses from an SSE endpoint, that contain a
- * <tt>{@value javax.ws.rs.core.HttpHeaders#RETRY_AFTER}</tt> HTTP header with a valid value. The
- * <tt>HTTP 503 + {@value javax.ws.rs.core.HttpHeaders#RETRY_AFTER}</tt> technique is often used by HTTP endpoints
- * as a means of connection and traffic throttling.
- * In case a <tt>HTTP 503 + {@value javax.ws.rs.core.HttpHeaders#RETRY_AFTER}</tt> response is received in return to a connection
- * request, JAX-RS SSE event source will automatically schedule a new reconnect attempt and use the received
- * <tt>{@value javax.ws.rs.core.HttpHeaders#RETRY_AFTER}</tt> HTTP header value as a one-time override of the reconnect delay.
+ * In addition to handling the standard connection loss failures, JAX-RS {@code SseEventSource} behaves differently to various
+ * HTTP response status codes and headers:
+ * <ul>
+ * <li>200 - with <tt>{@value javax.ws.rs.core.HttpHeaders#CONTENT_TYPE}</tt> header of "text/event-stream": This is normal operation. <code>onEvent</code> is invoked for each event.
+ *   <code>onComplete</code> is invoked when there are no more events. <code>onError</code> is only invoked if a non-recoverable error
+ *   occurs during processing.</li>
+ * <li>200 - with unsupported or missing <tt>{@value javax.ws.rs.core.HttpHeaders#CONTENT_TYPE}</tt> header: This is an error condition. <code>onError</code> and <code>onComplete</code> should be invoked.</li>
+ * <li>204: This indicates that server has no events to send. Only <code>onComplete</code> should be invoked.</li>
```

Comment:
Shouldn't `204` cleary imply that the result is not an empty stream but `null`?

Reply:
As you mentioned the HTML5 spec, can you please point me to the particular chapter where 204 in mentioned (maybe they use it in a specific way for SSE)? Thanks. :-)

Reply:
https://html.spec.whatwg.org/multipage/server-sent-events.html#server-sent-events

> Clients will reconnect if the connection is closed; a client can be told to stop reconnecting using the HTTP 204 No Content response code.

Reply:
I would propose to instead write:
`204: This indicates that the client must close the connection and must not reconnect. <code>onComplete</code> must be invoked. <code>onError</code> must not be invoked.`

Reply:
I strongly agree with @mkarg here. The new javadoc is actually weakens the specification. Any SHOULD can not be tested, and any implementation can behave on its will. 

> the client should automatically attempt

> consumer <i>should</i> be a WebApplicationException